src/transformers/models/bloom/modeling_bloom.py

```diff
@@ -0,0 +1,958 @@
+                )
+        else:
+            output_tensor = nn.functional.linear(context_layer, self.dense.weight)
+
+        output_tensor = output_tensor
+        output_bias = self.dense.bias
+
+        output = output_tensor.transpose(1, 0)
+        outputs = (output, present)
+        if output_attentions:
+            outputs += (attention_probs,)
+
+        return outputs, output_bias
+
+
+class BloomMLP(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+        hidden_size = config.hidden_size
+
+        self.pretraining_tp = config.pretraining_tp
+        self.slow_but_exact = config.slow_but_exact
+        self.activation_func = bias_gelu_impl
+        self.dense_h_to_4h = nn.Linear(hidden_size, 4 * hidden_size)
+        self.dense_4h_to_h = nn.Linear(4 * hidden_size, hidden_size)
+
+    def forward(self, hidden_states):
+        input_ = hidden_states
+
+        hidden_states = self.activation_func(
+            nn.functional.linear(hidden_states, self.dense_h_to_4h.weight), self.dense_h_to_4h.bias
+        )
+
+        if self.pretraining_tp > 1 and self.slow_but_exact:
+            intermediate_output = torch.zeros_like(input_)
+            slices = self.dense_4h_to_h.weight.shape[-1] / self.pretraining_tp
+            for i in range(self.pretraining_tp):
+                intermediate_output = intermediate_output + nn.functional.linear(
+                    hidden_states[:, :, int(i * slices) : int((i + 1) * slices)],
+                    self.dense_4h_to_h.weight[:, int(i * slices) : int((i + 1) * slices)],
+                )
+        else:
+            intermediate_output = nn.functional.linear(hidden_states, self.dense_4h_to_h.weight)
+
+        output = intermediate_output
+        output_bias = self.dense_4h_to_h.bias
+
+        return output, output_bias
+
+
+class BloomBlock(nn.Module):
+    def __init__(self, config, layer_number=None):
+        super().__init__()
+        hidden_size = config.hidden_size
+
+        self.input_layernorm = LayerNorm(hidden_size, eps=config.layer_norm_epsilon)
+        self.n_head = config.n_head
+        self.self_attention = BloomAttention(config, layer_number=layer_number)
+        self.post_attention_layernorm = LayerNorm(hidden_size, eps=config.layer_norm_epsilon)
+
+        self.mlp = BloomMLP(config)
+
+        self.apply_residual_connection_post_layernorm = config.apply_residual_connection_post_layernorm
+        self.bias_dropout_fusion = config.bias_dropout_fusion
+        self.hidden_dropout = config.hidden_dropout
+
+    def forward(
+        self,
+        hidden_states,
+        layer_past=None,
+        attention_mask=None,
+        head_mask=None,
+        use_cache=False,
+        output_attentions=False,
+    ):
+        # hidden_states: [batch_size, seq_length, hidden_size]
+
+        # Layer norm at the beginning of the transformer layer.
+        layernorm_output = self.input_layernorm(hidden_states)
+
+        # Compute alibi tensor
```

Comment:
what is an alibi tensor? 

What is it's use? Would be great if we could document this a bit more actually